Exercices/hangman.py

```diff
@@ -0,0 +1,232 @@
+             __________
+             |
+             |
+             |
+             |
+             |
+            _|_______
+        ___/         \___
+        '''
+
+        self.versuch7 = '''
+             __________
+             |/
+             |
+             |
+             |
+             |
+            _|_______
+        ___/         \___
+        '''
+
+        self.versuch6 = '''
+             __________
+             |/        |
+             |
+             |
+             |
+             |
+            _|_______
+        ___/         \___
+        '''
+
+        self.versuch5 = '''
+             __________
+             |/        |
+             |         O
+             |
+             |
+             |
+            _|_______
+        ___/         \___
+        '''
+
+        self.versuch4 = '''
+             __________
+             |/        |
+             |         O
+             |         |
+             |
+             |
+            _|_______
+        ___/         \___
+        '''
+
+        self.versuch3 = '''
+             __________
+             |/        |
+             |         O
+             |        -|
+             |
+             |
+            _|_______
+        ___/         \___
+        '''
+
+        self.versuch2 = '''
+             __________
+             |/        |
+             |         O
+             |        -|
+             |          \ 
+             |
+            _|_______
+        ___/         \___
+        '''
+
+        self.versuch1 = '''
+             __________
+             |/        |
+             |         O
+             |        -|-
+             |          \ 
+             |
+            _|_______
+        ___/         \___
+        '''
+
+        self.versuch0 = '''
+             __________
+             |/        |
+             |         O
+             |        -|-
+             |        / \ 
+             |  GAME OVER
+            _|_______
+        ___/         \___
+        '''
+
+    def print_hangman(self, number):
+        match number:
+            case 10:
+                print(self.versuch10)
+            case 9:
+                print(self.versuch9)
+            case 8:
+                print(self.versuch8)
+            case 7:
+                print(self.versuch7)
+            case 6:
+                print(self.versuch6)
+            case 5:
+                print(self.versuch5)
+            case 4:
+                print(self.versuch4)
+            case 3:
+                print(self.versuch3)
+            case 2:
+                print(self.versuch2)
+            case 1:
+                print(self.versuch1)
+            case 0:
+                print(self.versuch0)
+
+    def wortlaenge(self):
+        self.print_hangman(self.versuche)
+
+        for buchstabe in self.wort:
+            if buchstabe.lower() in self.gefunden:
+                print(buchstabe, end=" ")
+            else:
+                print("_ ", end=" ")
+
+        print("\n")
```

Comment:
`wortlaenge` sagt noch nicht wirklich viel darüber aus, was diese Methode genau macht. Ich würde eine Unterteilung vorschlagen:

```suggestion
    def show_letters(self):
        for buchstabe in self.wort:
            if buchstabe.lower() in self.gefunden:
                print(buchstabe, end=" ")
            else:
                print("_ ", end=" ")
        print("\n")
 
    def show(self):
        self.show_letters()
        self.print_hangman()
```

(Falls Sie lieber deutsche Methoden-Namen wollen:
- zeige_buchstaben
- zeige_wort
- zeige_gefunden
- ...?

--> Konsistent entweder deutsch oder englisch brauchen, möglichst kein Misch-Masch (fällt mir auch ab und zu schwer)
)